Modify /compiler-info to show both active and inactive compilers

azu-b opened the pull request:
### Why make this change?

We need an endpoint that returns all the compilers, both active and inactive. Before this change, endpoints only show active compilers.

### What is the change?

It adds the inactive compilers and their `not built` status to the compilers list returned by the `/compiler-info` endpoint.

### Testing

Run `yarn test` and make sure all tests pass. 

### Questions and additional information for Reviewer
First time contributing to Kevin 👀

azu-b committed:
Save all config names in Kevin constructor

azu-b committed:
Add inactive compilers to the getAllCompilerInfo result

azu-b committed:
Create a simple test for getAllCompilerInfo

azu-b committed:
Add another test for getAllCompilerInfo

joebeachjoebeach commented on lib/CompilerManager.js:
```diff
@@ -143,13 +143,30 @@ class CompilerManager {
      * @return {Object} a map of data keyed on compiler names
      */
-    getAllCompilerInfo() {
+    getAllCompilerInfo(configNames) {
```
The compiler manager is responsible for managing active compilers only. Augmenting this method to tell us information about inactive compilers by passing in all compilers mixes concerns, which isn't ideal. I think it would be best to move the logic for incorporating inactive compilers into the result to the `if (reqPath === '${this.kevinApiPrefix}/compiler-info') {` conditional in `middleware.js`.

```js
const activeCompilerInfo = manager.getAllCompilerInfo();
this.configs.forEach(({ name }) => {
    if (activeCompilerInfo.compilers.hasOwnProperty(name)) {
        return;
    }
    activeCompilerInfo.compilers[name] = { status: NOT_BUILT }
});
```

joebeachjoebeach commented on test/lib/CompilerManager.test.js:
```diff
@@ -118,3 +124,85 @@ describe("getAllBuildStatuses", () => {
     });
 });
+
+describe("getAllCompilerInfo", () => {
```
nice tests

azu-b committed:
Return CompilerManager and middleware to their original state

This also keeps the tests I made, but I removed the part that involved
inactive compilers

azu-b committed:
Add inactive compilers to response for compiler-info

azu-b commented on lib/CompilerManager.js:
```diff
@@ -147,7 +147,7 @@ class CompilerManager {
     getAllCompilerInfo() {
         const result = { compilers: {}, leastUsedCompiler: null };
-        Object.entries(this.activeCompilers).forEach(([name, compiler]) => {
+        Object.entries(this.activeCompilers).forEach(([name, _compiler]) => {
```
I made this change because Visual Studio said I didn't use `compiler` anywhere in that file, and I was kind of annoyed by the message 😅 , I can remove it if you want.

mq2thez replied:
I think you can switch to ```Object.keys(this.activeCompilers).forEach((name) => {
...
});```

azu-b committed:
Refactor a line due to ESLint rules
